src/nupic/math/ArrayAlgo.hpp

```diff
@@ -75,7 +75,7 @@ namespace nupic {
 
   // TODO: add asm code for gcc/clang/... on Windows
 
-  #elif defined(NTA_PLATFORM_darwin86) || defined(NTA_PLATFORM_linux32)
```

Comment:
@breznak This will break the future builds!

```
#if defined(NTA_PLATFORM_win32) && defined(NTA_COMPILER_MSVC)
...
#elif defined(NTA_ARCH_32)
...
#endif
```

If by any of the conditions in `#if defined(NTA_PLATFORM_win32) && defined(NTA_COMPILER_MSVC)` is false, the compiler will can process non-Windows ASM in `#elif defined(NTA_ARCH_32)`!

Maybe you should use:
`#elif !defined(NTA_PLATFORM_win32) && defined(NTA_ARCH_32)`
defined(NTA_PLATFORM_win32)